### ISSUE TITLE
Unclear how to use nuxt 2.9 programmaticly with typescript 

### ISSUE THREAD

**Issue by gotenxds** [nuxt/typescript#139]
### What problem does this feature solve?

Hello, I use nuxt 2.8 with typescript and koa, so far I was able to get things to work using 
``` cross-env NODE_ENV=development ts-node-dev -P ./server/tsconfig.json --transpile-only server/index.ts --watch server```
for my dev script
and ``` nuxt build && tsc -p ./server/ ``` for my build script

With nuxt 2.9 I have to use nuxt-ts to make nuxt compile my vuex modules, if I dont I get this error: 
``` [vuex] module namespace not found in mapGetters(): ```
But if I use nuxt-ts then my server wont run, how am I suppose to do this correctly?

### What does the proposed changes look like?

Add some documentation about running nuxt problematically.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9844">#c9844</a>)</em></sub></div>

**Comment by gotenxds** [comment]
Any info on this ?

**Comment by kevinmarrec** [comment]
I'm overall looking for help from people that already have setuped projects with programmatically TypeScript. I don't use programmatical usage and don't really recommend it. Having seperated API and/or use Proxy https://github.com/nuxt-community/proxy-module

**Comment by gotenxds** [comment]
Hi, so I took me around 30m to move my setup from running progrematicly to running with the proxy module so no biggy, BUT I do think that if running progrematicly is not recommended then it should say that in this page https://nuxtjs.org/api/nuxt/

**Comment by georgyfarniev** [comment]
Yes, I agree there is should be a note about it. If nuxt exposes api for this and provides scaffolding for it (create-nuxt-app option is programmatic usage), then there should be some clarity about it.

@kevinmarrec why is programmatic usage are not recommended by the way?

**Comment by kevinmarrec** [comment]
Programmatic usage is documented, but not for TypeScript.
Overall all internals of nuxt aren't typed, which means `Nuxt` & `Builder` classes

> @kevinmarrec why is programmatic usage are not recommended by the way?

Overall it add complexity and people asking for help, meanwhile more convenient solutions may/must fit your use cases.

I highly recommend a blog post from one of our core team members @manniL : https://blog.lichter.io/posts/nuxt-with-an-api

Having your server both delivering vue app & serving api means you're limiting your application to be a **Monolith**, which in other means : one single block, indivisible, slow/hard to change.

Separating your frontend (Nuxt) and your backend (API) makes thing easier in several fields :
- Recruitment: focus on recruting specialists in frontend (Vue.js/Nuxt), or backend (Node.js, Laravel, etc...)
- Maintenance : Independent versioning and separation of concerns just make things easier
- Performance : If your server server API requests as well as SSR Nuxt renders, it may happen it results in really slow page display when API is heavily requested. It also comes with resources used by your servers. With a monolith it will become really hard to know if a RAM issue is cause by Nuxt or your Express (or other lib) server. 

There are probably many other advantages but won't enumerate all

Still, Nuxt programmatic usage can answer to some use cases i'm not aware of OR not experienced in.

**Comment by georgyfarniev** [comment]
@kevinmarrec thank you for detailed response. So there’s no plan to deprecate rendering api, just not well supported with ts, right?

I saw this article. The problem that I wanted to implement auth on nuxt middleware level (not allow unauthorized people to get static assets with app code). Also I wanted to use session for nuxt all, not implement sophisticated authentication. Nuxt auth module not fit for my case because I used very custom sso auth. 

So I have to deal with serverMiddleware that based on connect, that’s a problem. It has very old deps and doesn’t have convenient middlewares. Even query string parser is a separate package. 

I still don’t understand why creators of nuxt want to make custom server instead of using koa (metal), what can be more lightweight than it? I want to write my thoughts and discuss with nuxt team later, one more time. Maybe I miss something, but koa seems to be best solution even for same arguments as used in defence of connect usage instead.

**Comment by Fnxxxxo** [comment]
@georgyfarniev Maybe the Nuxt team consider serverMiddleware as a middeware and just chain them. If so, there is no need to nest middeware and use Connect to keep it simple.

**Comment by georgyfarniev** [comment]
@Fnxxxxo serverMiddleware and middleware for nuxt are different things, or I misunderstood your question.

**Comment by b-m-f** [comment]
What we are using 
```
import {
  Nuxt,
  Builder,
} from 'nuxt'

const app = express();
const nuxt = new Nuxt(config);
await nuxt.ready();


const {
    host = process.env.HOST || process.env.NUXT_HOST || '127.0.0.1',
    port = process.env.PORT || process.env.NUXT_PORT || 3000,
} = nuxt.options.server;

if (config.dev) {
    const builder = new Builder(nuxt);
    await builder.build();
}
app.use(nuxt.render)
```

**Comment by chanon** [comment]
I was able to make it kind of work with code like b-m-f's above, but in the end I ran into this issue:
https://github.com/nuxt/nuxt.js/issues/5508

Which requires module esnext configuration in tsconfig to fix, which then blows up node server side code that does import * from ....

So I gave up and just use single page app.

**Comment by kevinmarrec** [comment]
Closed by #233
